### PR TITLE
Target Java 8

### PR DESCRIPTION
**Pull request by jimmyjames**
### Changes

This change sets the minimum required Java version for this library to Java 8.

While this library itself targets the Java 7 platform, third-party dependency updates have effectively made Java 8 the minimum required version since version 1.0.7.

As Java 7 is no longer receiving public updates, this change explicitly targets Java 8. 

Additionally, this will allow us to take advantage of features available in Java 8, such as new Date/Time APIs, better asynchronous support, among other features, going forward.

### Checklist

- [X] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [X] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [X] All existing and new tests complete without errors

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -40,25 +54,12 @@ test {
...
-        classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.8.4'
-        classpath 'gradle.plugin.com.auth0.gradle:oss-library:0.10.0'
-    }
-}
-
-repositories {
-    jcenter()
-}
-
 dependencies {
```
Comment (jimmyjames) [inline build.gradle:57]: The addition of these dependencies aren't new dependencies, they are just explicitly added now. Gradle 4 was allowing the usage of these transient dependencies through other dependencies, however when updating to Gradle 6 it did not. I did consider not adding the commons-codec dependency (used for Base64 encoding), and instead using the built-in Java 8 functionality, but decided that's something that should be done in a separate PR so we can test/verify.

File: lgtm.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+##########################################################################################
+
+extraction:
+  # Define settings for Java analysis
+  ####################################
+  java:
+    # The `index` step extracts information from the files in the codebase.
+    index:
+      # Specify the Java version required to build the project.
+      java_version: 1.8
```
Comment (lbalmaceda) [inline lgtm.yml:19]: According to the docs, that should be the default value. So this change shouldn't be required anyway. 

What error were you receiving before?

image

File: build.gradle [hunk 3]
```diff
@@ -69,4 +70,5 @@ dependencies {
     testImplementation 'org.mockito:mockito-core:2.8.9'
     testImplementation 'junit:junit:4.12'
     testImplementation 'org.springframework:spring-test:4.3.14.RELEASE'
+    testImplementation 'com.squareup.okhttp3:okhttp:4.9.0'
```
Comment (lbalmaceda) [inline build.gradle:73]: why are these 3 dependencies required now?

File: build.gradle [hunk 4]
```diff
@@ -1,7 +1,15 @@
-group 'com.auth0'
+plugins {
+    id 'java'
+    id 'jacoco'
+    id 'com.auth0.gradle.oss-library.java'
+    id "com.jfrog.bintray" version "1.8.4"
```
Comment (lbalmaceda) [inline build.gradle:5]: Please move the version to the `pluginManagement` block.

```suggestion
    id 'com.jfrog.bintray'
```

File: settings.gradle [hunk 5]
```diff
@@ -1,2 +1,15 @@
-rootProject.name = 'mvc-auth-commons'
+pluginManagement {
+    repositories {
+        gradlePluginPortal()
+        jcenter()
+        maven {
+            name "JCenter Gradle Plugins"
```
Comment (lbalmaceda) [inline settings.gradle:6]: Did you test if this custom maven repository is required?

File: settings.gradle [hunk 6]
```diff
@@ -1,2 +1,15 @@
...
+            url "https://dl.bintray.com/gradle/gradle-plugins"
+        }
+    }
+    plugins {
+        id 'com.auth0.gradle.oss-library.java' version '0.12.1'
```
Comment (lbalmaceda) [inline settings.gradle:11]: add a line for the jfrog one:

```
        id 'com.jfrog.bintray' version '1.8.5'
```

**Review by lbalmaceda — CHANGES_REQUESTED** [review]

File: settings.gradle (hunk 5 above)
Comment (jimmyjames) [reply settings.gradle]: you are right; not needed so removed.

File: build.gradle (hunk 4 above)
Comment (jimmyjames) [reply build.gradle]: Updated as suggested.

File: settings.gradle (hunk 6 above)
Comment (jimmyjames) [reply settings.gradle:7]: Done!

File: lgtm.yml (hunk 2 above)
Comment (jimmyjames) [reply lgtm.yml]: Ultimately we're running into this bug as @lbalmaceda discovered: https://github.com/gradle/gradle/issues/15049. I've removed the toolchain config and the lgtm.yml config.

File: build.gradle (hunk 3 above)
Comment (jimmyjames) [reply build.gradle:67]: See my comment above: https://github.com/auth0/auth0-java-mvc-common/pull/68#discussion_r520698297

**Review by lbalmaceda — APPROVED** [review]

**Pull request merged**